internal/function/blame.go

```diff
@@ -0,0 +1,145 @@
+package function
+
+import (
+	"fmt"
+	"github.com/src-d/gitbase"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+)
+
+type (
+	// Blame implements git-blame function as UDF
+	Blame struct {
+		repo   sql.Expression
+		commit sql.Expression
+	}
+
+	// BlameLine represents each line of git blame's output
+	BlameLine struct {
+		Commit  string `json:"commit"`
+		File    string `json:"file"`
+		LineNum int    `json:"linenum"`
+		Author  string `json:"author"`
+		Text    string `json:"text"`
+	}
+)
+
+// NewBlame constructor
+func NewBlame(repo, commit sql.Expression) sql.Expression {
+	return &Blame{repo, commit}
+}
+
+func (b *Blame) String() string {
+	return fmt.Sprintf("blame(%s, %s)", b.repo, b.commit)
+}
+
+// Type implements the sql.Expression interface
+func (*Blame) Type() sql.Type {
+	return sql.Array(sql.JSON)
+}
+
+// TransformUp implements the sql.Expression interface
+func (b *Blame) TransformUp(fn sql.TransformExprFunc) (sql.Expression, error) {
+	repo, err := b.repo.TransformUp(fn)
+	if err != nil {
+		return nil, err
+	}
+
+	commit, err := b.commit.TransformUp(fn)
+	if err != nil {
+		return nil, err
+	}
+
+	return fn(&Blame{repo, commit})
+}
+
+// Children implements the Expression interface.
+func (b *Blame) Children() []sql.Expression {
+	return []sql.Expression{b.commit}
+}
+
+// IsNullable implements the Expression interface.
+func (*Blame) IsNullable() bool {
+	return false
```

Comment:
you should check if children are nullable instead of deciding that here.

Reply:
Is this actually resolved?

Reply:
It was... must have been some rebase/branch fuckup. Fixed.